server/core/include/irods/irods_resource_constants.hpp

```diff
@@ -72,6 +72,7 @@ namespace irods
     extern const std::string RESOURCE_OP_RESOLVE_RESC_HIER;
     extern const std::string RESOURCE_OP_REBALANCE;
     extern const std::string RESOURCE_OP_NOTIFY;
+    extern const std::string RESOURCE_OP_READ_CHECKSUM;
```

Comment:
Maybe this should be `RESOURCE_OP_READ_CHECKSUM_FROM_STORAGE_DEVICE`?

That makes it more specific and gives us more wiggle room for future checksum operations to be added to the resource plugins.

Reply:
Done - also in S3 plugin.